app/controllers/components/course/groups_component.rb

```diff
@@ -12,4 +14,14 @@ def sidebar_items
       }
     ]
   end
+
+  private
+
+  def can_manage_group?
+    can?(:manage, Course::Group.new(course: current_course)) || manageable_groups.any?
+  end
+
+  def manageable_groups
+    current_course.groups.select { |group| can?(:manage, group) }
```

Comment:
it should be the same right? accessible_by is filtering a set of candidate rows to one that matches the condition?